tests/test_planning.py

```diff
@@ -54,24 +57,34 @@ def test_task_plan():
     heuristic = utils.create_task_planning_heuristic("hadd", init_atoms,
                                                      task.goal, ground_nsrts,
                                                      env.predicates, objects)
-    skeleton, _, _ = task_plan(init_atoms,
-                               task.goal,
-                               ground_nsrts,
-                               reachable_atoms,
-                               heuristic,
-                               timeout=1,
-                               seed=123)
+    task_plan_generator = task_plan(init_atoms,
+                                    task.goal,
+                                    ground_nsrts,
+                                    reachable_atoms,
+                                    heuristic,
+                                    timeout=1,
+                                    seed=123)
+    skeleton, _, metrics = next(task_plan_generator)
     assert len(skeleton) == 2
     assert isinstance(skeleton[0], _GroundNSRT)
     assert isinstance(skeleton[1], _GroundNSRT)
+    total_num_skeletons = 1
+    with pytest.raises(ApproachFailure) as e:
```

Comment:
just curious, do you have any idea what these 3 skeletons are?

Reply:
well the first one is what we'd expect (a length 2 optimal plan), and then the second two are length 4 plans that pick and place unnecessarily.